arch/risc-v/src/common/riscv_macros.S

```diff
@@ -84,6 +85,69 @@
   REGSTORE   x30, REG_X30(\in)   /* t5 */
   REGSTORE   x31, REG_X31(\in)   /* t6 */
 
+  csrr       s0, CSR_STATUS
```

Comment:
I would personally keep the status register and EPC handling separate. 

Three things are needed:
- Integer register handling
- Status handling
- FPU handling